pkg/webhook/cluster/mutating_webhook.go

```diff
@@ -30,10 +30,10 @@ func (a *ManagedClusterMutatingAdmissionHook) MutatingResource() (schema.GroupVe
 }
 
 // Admit is called by generic-admission-server when the registered REST resource above is called with an admission request.
-func (a *ManagedClusterMutatingAdmissionHook) Admit(req *admissionv1beta1.AdmissionRequest) *admissionv1beta1.AdmissionResponse {
+func (a *ManagedClusterMutatingAdmissionHook) Admit(req *admissionv1.AdmissionRequest) *admissionv1.AdmissionResponse {
```

Comment:
I think you want to have this release of your admission webhook handle both v1beta1 admissionreview  and v1 admissionreview.  Allowing both will allow you to update the webhook server first and the admission registration second.  Probably in two different releases.

1. release current - registration v1beta1 review, server v1beta1 review
2. current +1 - registration v1beta1 review, server v1 and v1beta1 review
3. current +2 - registration v1 review, server v1 and v1beta1 review
4. current +3 - registration v1 review, server v1 review